src/Centreon/Domain/HostConfiguration/UseCase/V21/HostGroup/FindHostGroups.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\HostConfiguration\UseCase\V21\HostGroup;
+
+use Centreon\Domain\HostConfiguration\Exception\HostGroupException;
+use Centreon\Domain\HostConfiguration\Interfaces\HostGroup\HostGroupReadRepositoryInterface;
+use Centreon\Domain\HostConfiguration\Model\HostGroup;
+
+/**
+ * This class is designed to represent a use case to find all host groups
+ *
+ * @package Centreon\Domain\HostConfiguration\UseCase
+ */
+class FindHostGroups
+{
+    /**
+     * @var HostGroupReadRepositoryInterface
+     */
+    private $configurationReadRepository;
```

Comment:
IMO, i found this naming a little confusing, why not called this  `$hostGroupReadRepository` ?